### PR TITLE
quarkus-cli should allow you to select which java version to target

### PR DESCRIPTION
**Pull request by pjgg**
### Summary

Quarkus-cli should be able to select which java version to target. This PR verifies this behavior

Please select the relevant options.

- [X] New scenario (non-breaking change which adds functionality)

### Checklist:
- [X] Methods and classes used in PR scenarios are meaningful
- [X] Commits are well encapsulated and follow [the best practices](https://cbea.ms/git-commit/)

### CODE REVIEW

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java [hunk 1]
```diff
@@ -210,4 +229,8 @@ private File getPomFileFromMavenApplication(QuarkusCliRestService app) {
                 .findFirst()
                 .orElseThrow(() -> new RuntimeException("Malformed Maven Quarkus application. Missing pom.xml"));
     }
+
+    private String truncateCurrentJavaVersion() {
```
Comment (jsmrcka) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java:233]: The name of this method does not correspond with what it does.

And what is the point here? To use JDK which is not the default in the system?
I don't see why. CLI's `--java` supports `11` and `17` (`11` is default), so why not just try both options, regardless to the system default?

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java [hunk 2]
```diff
@@ -62,7 +64,25 @@ public void shouldCreateApplicationOnJvm() {
...
+    @Tag("QUARKUS-1472")
+    @Test
+    public void shouldCreateAnApplicationForcingJavaVersion() {
+
+        // Create application
+        QuarkusCliClient.CreateApplicationRequest args = defaults().withExtraArgs("--java=" + truncateCurrentJavaVersion());
+        QuarkusCliRestService app = cliClient.createApplication("app", args);
+
+        // Should build on Jvm
+        QuarkusCliClient.Result result = app.buildOnJvm();
```
Comment (jsmrcka) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java:79]: I think this won't work unless we make sure that `maven-compiler-plugin` uses the same JDK as the one provided by the `--java` param.
Is that even possible? We would need to make sure the JDK is installed, and also specify `JAVA_HOME`.

**Review by jsmrcka — CHANGES_REQUESTED** [review]

**Review by jsmrcka — APPROVED** [review]

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java [hunk 3]
```diff
@@ -188,13 +212,13 @@ private void assertInstalledExtensions(QuarkusCliRestService app, String... expe
...
         MavenXpp3Reader reader = new MavenXpp3Reader();
-        String javaVersion = getSystemJavaVersion();
         try {
             Model model = reader.read(new FileReader(pomFile));
-            Assertions.assertEquals(model.getProperties().get("maven.compiler.release"), javaVersion,
-                    "Unexpected Java version. Java support tool should detect host Java version");
+            Assertions.assertEquals(model.getProperties().get("maven.compiler.release"), expectedJavaVersion,
+                    "Unexpected Java version. Java support tool should detect host Java version or use " +
+                            "the provided one by --java argument");
         } catch (IOException | XmlPullParserException e) {
```
Comment (rsvoboda) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java:222]: Also check generated Dockerfile.jvm. This is needed for good OpenShift experience.

File: quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java [hunk 4]
```diff
@@ -188,13 +212,13 @@ private void assertInstalledExtensions(QuarkusCliRestService app, String... expe
...
-    private void assertExpectedJavaVersion(File pomFile) {
+    private void assertExpectedJavaVersion(File pomFile, String expectedJavaVersion) {
         MavenXpp3Reader reader = new MavenXpp3Reader();
-        String javaVersion = getSystemJavaVersion();
         try {
             Model model = reader.read(new FileReader(pomFile));
-            Assertions.assertEquals(model.getProperties().get("maven.compiler.release"), javaVersion,
-                    "Unexpected Java version. Java support tool should detect host Java version");
+            Assertions.assertEquals(model.getProperties().get("maven.compiler.release"), expectedJavaVersion,
+                    "Unexpected Java version. Java support tool should detect host Java version or use " +
```
Comment (rsvoboda) [inline quarkus-cli/src/test/java/io/quarkus/ts/quarkus/cli/QuarkusCliCreateJvmApplicationIT.java:220]:
```suggestion
                    "Unexpected Java version defined in maven.compiler.release property of pom.xml. Java support tool should detect host Java version or use " +
```

**Review by rsvoboda — CHANGES_REQUESTED** [review]

**Review by rsvoboda — APPROVED** [review]

**Pull request merged**